src/proto/meta_protocol.rs

```diff
@@ -200,4 +212,52 @@ impl MetaProtocol for Client {
                 .transpose(),
         }
     }
+
+    async fn meta_arithmetic<K: AsRef<[u8]>>(
+        &mut self,
+        key: K,
+        meta_flags: Option<&[&str]>,
+    ) -> Result<Option<MetaValue>, Error> {
+        let kr = Self::validate_key_length(key.as_ref())?;
+
+        self.write_non_set_command("ma", kr, meta_flags).await?;
+
+        self.conn.flush().await?;
+
+        match self.drive_receive(parse_meta_arithmetic_response).await? {
+            MetaResponse::Status(Status::Stored) => Ok(None),
+            MetaResponse::Status(Status::NoOp) => Ok(None),
+            MetaResponse::Status(s) => Err(s.into()),
+            MetaResponse::Data(d) => d
+                .map(|mut items| {
+                    let item = items.remove(0);
+                    Ok(item)
+                })
+                .transpose(),
+        }
+    }
+
+    async fn write_non_set_command(
```

Comment:
Is there something that groups this that isn't a negative? Like all `retrieval` commands? I think that is how memcached organizes them https://github.com/memcached/memcached/blob/master/doc/protocol.txt#L62

Reply:
Actually, nevermind.  Irrelevant if I'm walking that bit of code back anyway 😅 .